script.js

```diff
@@ -1,4 +1,128 @@
+// gamemode , enter username name
+var gameMode = 'enter username mode';
+
+// store username
+var userNameOfThePlayer = '';
+
+// random number from 0 to 2
+var randomNum = function () {
+  var numZeroToTwo = Math.floor(Math.random() * 3);
+  return numZeroToTwo;
+};
+
+// user win
+var numOfUserWin = 0;
+
+// computer win
+var numOfComputerWin = 0;
+
+// total game
+var totalMatch = 0;
+
+// winning percentage
+var userWinPercentage = 0;
+var computerWinPercentage = 0;
+
+// number of draw
+var numOfDraw = 0;
+
+// assign value to words
+var assignNumberToWords = function () {
+  var numGenerator = randomNum();
+  if (numGenerator == 0) {
+    return 'scissors';
+  }
+  if (numGenerator == 1) {
+    return 'stone';
+  }
+  if (numGenerator == 2) {
+    return 'paper';
+  }
+  return 'got bug';
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue = 'Please enter one of the 3 options. scissors , paper , stone';
+
+  // do this if gamemode is 'enter username mode.
+  if (gameMode == 'enter username mode') {
+    // store input as username
+    userNameOfThePlayer = input;
+    // change game mode to start game.
+    gameMode = 'start game';
+    myOutputValue = 'Welcome ,' + userNameOfThePlayer
+    + ' . Please enter one of the 3 objects. scissors , paper , stone';
+    // if gamemode is start game  run this code
+  }
+  // if gamemode is start game run this code
+  else if (gameMode == 'start game') {
+    // if condition is true , result will either be draw , win or lose .
+    // Otherwise it will prompt user to enter the correct input
+    if ((input == 'scissors' || input == 'stone' || input == 'paper') == true) {
```

Comment:
`(input == 'scissors' || input == 'stone' || input == 'paper') == true` can be simplified to `input == 'scissors' || input == 'stone' || input == 'paper'`

Reply:
Currently your code doesn't do anything when input is not scissors, paper, or stone. Maybe you can return a prompt to the user if they enter invalid input?